dspace-server-webapp/src/test/java/org/dspace/app/rest/MetadatafieldRestRepositoryIT.java

```diff
@@ -168,6 +172,261 @@ public void findByNullSchema() throws Exception {
+            "http://www.dspace.org/ns/aschema2").build();
+
+        MetadataField metadataField = MetadataFieldBuilder
+            .createMetadataField(context, schema, "AnElement1", "AQualifier", "AScopeNote").build();
+
+        MetadataField metadataField2 = MetadataFieldBuilder
+            .createMetadataField(context, schema2, "AnElement2", "AQualifier", "AScopeNote2").build();
+
+        MetadataField metadataField3 = MetadataFieldBuilder
+            .createMetadataField(context, schema, "AnElement2", "AQualifier", "AScopeNote2").build();
+
+        context.restoreAuthSystemState();
+
+        super.runDSpaceScript("index-discovery", "-b");
+
+        getClient().perform(get(SEARCH_BYFIELDNAME_ENDPOINT)
+            .param("element", "AnElement2")
+            .param("qualifier", "AQualifier"))
+                   .andExpect(status().isOk())
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$._embedded.metadatafields", Matchers.hasItem(
+                       MetadataFieldMatcher.matchMetadataField(metadataField2))
+                                      ))
+                   .andExpect(jsonPath("$._embedded.metadatafields", Matchers.hasItem(
+                       MetadataFieldMatcher.matchMetadataField(metadataField3))
+                                      ))
+                   .andExpect(jsonPath("$._embedded.metadatafields", Matchers.not(hasItem(
+                       MetadataFieldMatcher.matchMetadataField(metadataField))
+                                                                                 )))
+                   .andExpect(jsonPath("$.page.size", is(20)))
+                   .andExpect(jsonPath("$.page.totalElements", is(2)));
+    }
+
+    @Test
+    public void findByFieldName_schemaAndQualifier() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        MetadataSchema schema = MetadataSchemaBuilder.createMetadataSchema(context, "ASchema",
+            "http://www.dspace.org/ns/aschema").build();
+        MetadataSchema schema2 = MetadataSchemaBuilder.createMetadataSchema(context, "ASchema2",
+            "http://www.dspace.org/ns/aschema2").build();
+
+        MetadataField metadataField = MetadataFieldBuilder
+            .createMetadataField(context, schema, "AnElement1", "AQualifier", "AScopeNote").build();
+
+        MetadataField metadataField2 = MetadataFieldBuilder
+            .createMetadataField(context, schema2, "AnElement2", "AQualifier", "AScopeNote2").build();
+
+        MetadataField metadataField3 = MetadataFieldBuilder
+            .createMetadataField(context, schema, "AnElement3", "AQualifier", "AScopeNote3").build();
+
+        context.restoreAuthSystemState();
+
+        super.runDSpaceScript("index-discovery", "-b");
+
+        getClient().perform(get(SEARCH_BYFIELDNAME_ENDPOINT)
+            .param("schema", schema.getName())
+            .param("qualifier", "AQualifier"))
```

Comment:
It looks like we're missing a test that uses `schema`, `element` and `qualifier` together to select a single *unique* MetadataField.  Could you add that in as well?